storage/src/mmr/bitmap.rs

```diff
@@ -0,0 +1,290 @@
+//! An authenticatable bitmap.
+
+use crate::mmr::{iterator::leaf_num_to_pos, mem::Mmr};
+use commonware_cryptography::Hasher as CHasher;
+use commonware_utils::SizedSerialize;
+
+/// A bitmap supporting inclusion proofs through Merkelization.
+///
+/// Merkelization of the bitmap is performed over chunks of DIGEST_SIZE bytes in order to reduce
+/// overhead (e.g. by a factor of >256 for 32-byte digests).
+pub struct Bitmap<H: CHasher> {
+    /// The bitmap itself, in chunks of size DIGEST_SIZE bytes. The number of valid bits in the last
+    /// chunk is given by `self.next_bit`.
+    bitmap: Vec<u8>,
+
+    /// The position within the last chunk of the bitmap where the next bit is to be appended.
+    ///
+    /// Invariant: This value is always in the range [0, DIGEST_SIZE * 8).
+    next_bit: u64,
```

Comment:
nit: consider if just using `usize` as the type here results in slightly less casting. Should be fine as the `CHUNK_SIZE` can never reach `usize`